### ISSUE TITLE
Follow an official design guideline HIG

### ISSUE THREAD

**Issue by bastiencyr** [borgbase/vorta#1050]
Hello, 

I started to read some guidelines and docs from official design guidelines 💪 . According to me,  vorta UI lacked consistency and didn't look modern but I couldn't say why. So after some research, I discovered Flat and Material design. Flat design is everywhere but I didn't realize and because I have always seen this theme (I'm rather young 😅 ), vorta UI interface seems really old to me.

Consistency is another notion and design guidelines are very helpful for that. If we want to make an interface that looks native, user should  find the elements in the same place and in the same way. For example, vorta doesnt follow everytime the header capitalization ( [Gnome](https://developer.gnome.org/hig/stable/guidelines/writing-style.html) , [Apple](https://developer.apple.com/design/human-interface-guidelines/macos/fields-and-labels/text-fields/) ). "One folder or file per line" should be "One Folder or File per Line" or "Can't delete the last profile." should be "Can't Delete the Last Profile". 

Since vorta users come from Apple, Gnome or KDE, it's more complicated. But lets be honest, all UI interfaces and guidelines converge towards the same ideas 😝 and looks the same !

So, my proposition is to follow KDE, Gnome and/or Apple guidelines as guide and to follow flat design. Note that Apple, Gnome and KDE have officially chosen the flat design. 

This is the three guidelines ❤️ : 
https://develop.kde.org/hig/introduction/
https://developer.gnome.org/hig/stable/principles.html
https://developer.apple.com/design/human-interface-guidelines/macos/app-architecture/onboarding/

There are many common points between these guidelines.

Thanks for reading 🥰

**Comment by m3nu** [comment]
Do you have some concrete examples that could be changed to follow such guidelines?

**Comment by bastiencyr** [comment]
> Do you have some concrete examples that could be changed to follow such guidelines?

I don't read the whole guidelines but this is some changes : 
For text : 

 - header capitalization as I said
 - limit the tooltip because touch screen can't print the tooltip
 -  keep text short (translation can expand the text to 30% more in some languges. For example translation of "Repositories and SSH keys are shared between profiles. Source folders, active destination repo, allowed networks, pruning, validation and scheduling are per-profile. (166 chars)" in french is "Les dépôt et les clés SSH sont partagés entre les profils. Les dossiers sources, le dépôt de destination actif, les réseaux autorisés, l'élagage, la validation et la planification sont propres à chaque profil. (210 char)". 
 - text must be selectable 
 -  adopt the software point of view and avoid pronouns like you. "Storing password in your password manager" -> "Storing password in the storing manager."
 - uniform font (in add profile, the font is 12... why ? In all the UI its 10). 
 - remove dot in misc
 - ...

Visual changes  : 
 - use no more than one discolure control by view (found in Apple guideline). For example, schedule tab contains 4 disclosures (clock, wifi, tasks, terminal).
 - use gradient buttons in view instead of separated buttons (https://developer.apple.com/design/human-interface-guidelines/macos/buttons/gradient-buttons/)
 - use a sidebar instead of tab views ( "Having more than six tabs can be overwhelming and create layout issues." -> https://developer.apple.com/design/human-interface-guidelines/macos/windows-and-views/tab-views/) 
 - ...

All this exemples are inspiring from Gnome and Apple guidelines. 
For a Flat look, you can see my last PR  in #1008 .

All this changes has interests for users. Its not just stupid rules and Vorta will look like native app.

**Comment by m3nu** [comment]
Those are all good points and even without reading the guidelines I see how they are good for consistency.

How about starting small and going larger over time:

1. You share a high level list of changes required. E.g. "Capitalize headers" here as checklist
2. Then you, me and others wanting to contribute can do small PRs to fix one item at a time. This will be better than one large PR.
3. When all the small items are cleaned up, we proceed to #1008, which has the biggest changes.

I prefer smaller changes because they are easy to review, revert and smaller tasks can be done by new contributors as well. We also have many translations attached to our text strings. If you change one, they all need to review their translations.

**Comment by bastiencyr** [comment]
> If you change one, they all need to review their translations.
 
 I do agree. We cant do that in just one PR, it will break every translation. 
 
> When all the small items are cleaned up, we proceed to UI Improvements #1008, which has the biggest changes.

Why ? #1008 just changes the interface, not the text.

**Comment by m3nu** [comment]
> Why ? #1008 just changes the interface, not the text.

True. Then just add it to the list as part of the interface work, so it's in context. No need to wait for text changes.

**Comment by bastiencyr** [comment]
Fine, I added a list in my first comment. I think we can add the label "Good first issue".
I will try to complete this list and quote a design guideline for each item.

**Comment by bastiencyr** [comment]
I also started a documentation about design UI vorta. It's just a compilation of non obvious ideas from several guidelines. Sometimes, I take some liberties but you can give your opinion if you don't agree. If a commit change the UI, it's good to check this.

**Comment by real-yfprojects** [comment]
> Put font 10 in profile add window.

I think this can be ticked since the text uses the html `body` tag which will automatically assign the right size (defined by the user/desktop environment).

**Comment by real-yfprojects** [comment]
> In source tab, add gradients bottoms '+/-' at bottom of the view.

This is how KDE implements gradient buttons in Qt:
https://develop.kde.org/hig/components/editing/list/#picker-1

**Comment by real-yfprojects** [comment]
The [KDE HIG](https://develop.kde.org/hig/style/writing/labels/#labels-for-checkboxes-and-radio-buttons) states regarding checkbox labels:
> use affirmative phrases rather than use negative phrases

This is not the case here (for example):
https://github.com/borgbase/vorta/blob/23c47673c009bdef8baebb0b9cdf5e78c07fe373/src/vorta/assets/UI/scheduletab.ui#L364

**Comment by real-yfprojects** [comment]
[KDE HIG](https://develop.kde.org/hig/style/writing/placeholder/):
> A placeholder with an example is not a replacement for a label, but should give the user only an additional hint about what kind of data is expected.

This is not the case in Scheduletab>shell commands.

**Comment by torwag** [comment]
May I ask/add why there is no (as common) context menu for the "Source" and the "Archives" tables? Source has buttons on the right, which is still easy to understand. However, in archives, you get some buttons below, and even this is split in a drop-down menu "Selected Archive" and three separate buttons.

This is particular difficult to get if you use displays in portrait orientation and a tiling window manager, as those buttons are really down at the bottom of an otherwise almost empty list. ;)

I also believe that the sub-menus are easier to get, if they would be located in a left-side window-pane as tree-view, as it can be found in many programs when it comes to configuration and settings.

**Comment by real-yfprojects** [comment]
The [KDE HIG](https://develop.kde.org/hig/components/navigation/contextmenu/) states:
> Don’t use context menus as the only way to access functionality. Every item in a context menu must be available via a method that is somehow visible by default

> Provide a context menu for any item with actions that can be performed on it.

So context menus should definitely be added to *vorta*'s item views.

**Comment by MarjaE2** [comment]
Is there an accessible version of the linked guidelines page? It has a very wide non-scrolling sidebar, so it triggers migraines.

**Comment by real-yfprojects** [comment]
> Are there accessible versions of any of the linked guidelines pages? All 3 have very wide non-scrolling sidebars, so all 3 trigger migraines.

I guess not.
